app/models/report.rb

```diff
@@ -32,24 +32,24 @@ class Report < ApplicationRecord
   has_many :report_elements, inverse_of: :report, dependent: :delete_all
 
   DEFAULT_SETTINGS = {
-    all_tasks: true,
+    all_tasks: 'true',
```

Comment:
Why did you change all booleans to strings?

Reply:
When we save it to the database it will become a string. And we need to compare it with `'true'` string.

Reply:
It doesn't

Reply:
When you create a new report it is ok. But when you load Report:
```
[1] pry(main)> Report.last.settings
  Report Load (1.7ms)  SELECT "reports".* FROM "reports" ORDER BY "reports"."id" DESC LIMIT $1  [["LIMIT", 1]]
=> {"all_tasks"=>"true",
 "task"=>
  {"protocol"=>{"description"=>"true", "completed_steps"=>"true", "uncompleted_steps"=>"true", "step_checklists"=>"true", "step_files"=>"true", "step_tables"=>"true", "step_comments"=>"true"},
   "file_results"=>"false",
   "file_results_previews"=>"false",
   "table_results"=>"true",
   "text_results"=>"true",
   "result_comments"=>"true",
   "result_order"=>"atoz",
   "activities"=>"true"}}
```

Reply:
Report.last.settings
  Report Load (0.7ms)  SELECT "reports".* FROM "reports" ORDER BY "reports"."id" DESC LIMIT $1  [["LIMIT", 1]]
=> {"all_tasks"=>true,
 "task"=>
  {"protocol"=>{"description"=>true, "completed_steps"=>true, "uncompleted_steps"=>true, "step_checklists"=>true, "step_files"=>true, "step_tables"=>true, "step_comments"=>true},
   "file_results"=>false,
   "file_results_previews"=>false,
   "table_results"=>true,
   "text_results"=>true,
   "result_comments"=>true,
   "result_order"=>"atoz",
   "activities"=>true}}

Reply:
Looks like an issue in params. In params, it is a string.

Reply:
So we need or convert params to boolean. Or change here to string.